contracts/hardhat/contracts/RewardPoolV3.sol

```diff
@@ -0,0 +1,256 @@
+     */
+    function claimReward(address account) external updateReward(account) {
+        uint256 reward = _rewards[account];
+        if (reward == 0) revert NoUnclaimedReward(account);
+
+        _rewards[account] = 0;
+        rewardToken.transfer(account, reward);
+
+        emit RewardClaimed(account, reward);
+    }
+
+    /**
+     * @notice Withdraws multiple stakes after the unstake period.
+     * @param to The address to receive unstaked tokens.
+     * @param stakeIds The IDs of the stakes to unstake.
+     */
+    function unstake(
+        address to,
+        uint256[] calldata stakeIds
+    ) external updateReward(msg.sender) {
+        uint256 totalAmount;
+        for (uint256 i; i < stakeIds.length; ++i) {
+            totalAmount += _processUnstake(stakeIds[i]);
+        }
+        stakingToken.transfer(to, totalAmount);
+    }
+
+    /**
+     * @notice Owner: Updates the unstake period duration.
+     * @param newUnstakePeriod The new unstake period duration in seconds.
+     */
+    function changeUnstakePeriod(uint32 newUnstakePeriod) external onlyOwner {
+        unstakePeriod = newUnstakePeriod;
+        emit UnstakePeriodChanged(newUnstakePeriod);
+    }
+
+    /**
+     * @notice Owner: Starts or updates a staking incentive program.
+     * @param extraReward Additional reward tokens to add to the program.
+     * @param incentiveDuration Duration of the incentive in seconds.
+     */
+    function updateStakingIncentive(
+        uint256 extraReward,
+        uint256 incentiveDuration
+    ) external onlyOwner updateReward(address(0)) {
+        if (
+            incentiveDuration < MIN_INCENTIVE_DURATION ||
+            incentiveDuration > MAX_INCENTIVE_DURATION
+        ) {
+            revert InvalidIncentiveDuration(
+                MIN_INCENTIVE_DURATION,
+                MAX_INCENTIVE_DURATION
+            );
+        }
+
+        uint256 reward = _remainingReward() + extraReward;
+        if (reward == 0) revert NoReward();
+
+        rewardRate = (reward * 1e18) / incentiveDuration;
+        updatedAt = uint32(block.timestamp);
+        incentiveEndsAt = uint32(block.timestamp + incentiveDuration);
+        _allTimeReward += uint96(extraReward);
+
+        rewardToken.transferFrom(msg.sender, address(this), extraReward);
+
+        emit StakingIncentiveUpdated(reward, incentiveEndsAt);
+    }
+
+    /**
+     * @notice Owner: Recovers excess tokens sent to the contract by mistake.
+     * @param token The ERC20 token to recover.
+     * @param to The recipient address.
+     */
+    function recover(IERC20 token, address to) external onlyOwner {
+        uint256 amount = token.balanceOf(address(this));
+        if (token == stakingToken) {
+            amount -= totalStaked;
+        } else if (token == rewardToken) {
+            amount -= _allTimeReward;
+        }
+
+        if (amount == 0) revert NoAvailableBalance(address(token));
+
+        token.transfer(to, amount);
```

Comment:
Propose to use OZ `SafeERC20` library for transfer unknown token

Reply:
fixed